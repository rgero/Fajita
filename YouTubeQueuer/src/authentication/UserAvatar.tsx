const UserAvatar = () => {
  const user = null;
 
  let avatar;
  if (!user)
  {
    avatar = "default-user.jpg"
  } else {
    //avatar = user.user_metadata["avatar"]
  }

  return (
    <div>
      <img className="w-10 h-10 rounded-full" src={avatar} alt="Avatar"/>
    </div>
  )
}

export default UserAvatar
