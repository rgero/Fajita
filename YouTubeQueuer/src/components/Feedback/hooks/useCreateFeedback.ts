import { submitFeedback } from "../../../services/apiGithubLogger";
import toast from "react-hot-toast";
import { useMutation, } from "@tanstack/react-query";

export const useCreateFeedback = () => {
  const {isPending: isAdding, mutate: addFeedback} = useMutation({
      mutationFn: ({title, description}: {title: string, description: string}) => submitFeedback({title, description}),
      onSuccess: () => {
          toast.success("Feedback Submitted");
      },
      onError: (err) => toast.error(err.message)
  })

  return {isAdding, addFeedback}
}