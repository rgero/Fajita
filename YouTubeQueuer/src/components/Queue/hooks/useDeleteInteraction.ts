import { useMutation, useQueryClient } from "@tanstack/react-query";

import { deleteFromQueue } from "../../../services/apiFajita";
import toast from "react-hot-toast";

export const useDeleteInteraction = () => {
    const queryClient = useQueryClient();
    const {isPending: isDeleting, mutate: deleteInteraction} = useMutation({
        mutationFn: (id: number) => deleteFromQueue(id),
        onSuccess: () => {
            queryClient.invalidateQueries({queryKey: ["queueList"]})
            toast.success("Video deleted!");
        },
        onError: (err) => toast.error(err.message)
    })

    return {isDeleting, deleteInteraction}
}