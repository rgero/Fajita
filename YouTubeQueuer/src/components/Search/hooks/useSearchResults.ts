import { YoutubeResponse } from "../../../interfaces/YoutubeResponse";
import { getSearchResults } from "../../../services/apiFajita";
import { useQuery } from "@tanstack/react-query";
import { useSearchParams } from "react-router-dom";

export interface SearchResponse {
  isLoading: boolean,
  searchResults: YoutubeResponse[],
  error: Error | null
}

export const useSearchResults = (): SearchResponse => {
  const [searchParams] = useSearchParams();
  const searchTerm:string|null = searchParams.get("search");
  const {isLoading, data: searchResults = [], error} = useQuery({queryKey: ["searchResults", searchTerm], queryFn: ()=>getSearchResults(searchTerm as string), refetchInterval: false});
  return { isLoading, searchResults, error};
}