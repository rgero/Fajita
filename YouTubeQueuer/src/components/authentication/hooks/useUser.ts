import { getCurrentUser } from "../../../services/apiAuthentication"
import { useQuery } from "@tanstack/react-query"

export const useUser = () => {
  const { isLoading, fetchStatus, data: user } = useQuery({
    queryKey: ["user"],
    queryFn: getCurrentUser,
  });
  
  const isAuthenticated: boolean = user ? user.id > 0 : false;
  return { isLoading, user, fetchStatus, isAuthenticated};
}