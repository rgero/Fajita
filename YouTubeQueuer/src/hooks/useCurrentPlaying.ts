import { YoutubeResponse } from "../interfaces/YoutubeResponse";
import { getCurrentPlaying } from "../services/apiFajita";
import { useQuery } from "@tanstack/react-query";

export const useCurrentPlaying = () => {
  const {isLoading, data: currentlyPlaying, error}: {isLoading: boolean, data: YoutubeResponse|undefined, error: Error|null} = useQuery({queryKey: ["current"], queryFn: () => getCurrentPlaying(), refetchInterval: 5000});
  return { isLoading, currentlyPlaying, error};
}