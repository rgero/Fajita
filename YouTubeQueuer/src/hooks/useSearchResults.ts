import { getSearchResults } from "../services/apiFajita";
import { useQuery } from "@tanstack/react-query";
import { useSearchParams } from "react-router-dom";

export const useSearchResults = () => {
  const [searchParams] = useSearchParams();
  const searchTerm:string|null = searchParams.get("search");
  const {isLoading, data: searchResults = [], error} = useQuery({queryKey: ["searchResults", searchTerm], queryFn: ()=>getSearchResults(searchTerm)});
  return { isLoading, searchResults, error};
}