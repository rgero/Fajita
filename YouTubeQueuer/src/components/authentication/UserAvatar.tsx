import { Avatar } from "@mui/material";
import { useDarkMode } from "../../context/DarkModeContext";
import { useUser } from "./hooks/useUser";

const UserAvatar = () => {
  const { toggleDarkMode } = useDarkMode();
  const {user} = useUser();

  // This should probably change to the First input Letter.
  return (
    <Avatar onClick={toggleDarkMode} >{user?.name[0]}</Avatar>  
  )
}

export default UserAvatar
