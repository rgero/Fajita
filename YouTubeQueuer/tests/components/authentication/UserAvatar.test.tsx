import { describe, it, vi } from 'vitest';

import React from "react";
import UserAvatar from '../../../src/components/authentication/UserAvatar';
import renderer from 'react-test-renderer';
import {useUser} from '../../../src/components/authentication/hooks/useUser';

// Mock the custom hook
vi.mock('../../../src/components/authentication/hooks/useUser');

vi.mock('react-router-dom', () => ({
  useNavigate: () => () => {},
}));

const testData = {
  email: "roy@roy.com",
  id: 1,
  name: "Roy"
}

describe('UserAvatar Tests', () => {
  it('renders data returned by useCustomHook', () => 
  {
    // Provide a mock implementation for the hook
    (useUser as jest.Mock).mockReturnValue(testData);

    const component = renderer.create(<UserAvatar/>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});