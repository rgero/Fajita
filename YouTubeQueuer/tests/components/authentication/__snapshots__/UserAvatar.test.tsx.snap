// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UserAvatar Tests > renders data returned by useCustomHook 1`] = `
<div
  className="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
  onClick={[Function]}
>
  <img
    className="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
    src="default-user.jpg"
  />
</div>
`;
